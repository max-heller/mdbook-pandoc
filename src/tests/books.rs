use std::path::{Path, PathBuf};

use once_cell::sync::Lazy;

use super::{Config, MDBook};

static BOOKS: Lazy<PathBuf> = Lazy::new(|| Path::new(env!("CARGO_MANIFEST_DIR")).join("books"));

#[test]
#[ignore]
fn mdbook_guide() {
    let logs = MDBook::load(BOOKS.join("mdBook/guide"))
        .config(Config {
            hosted_html: Some("https://rust-lang.github.io/mdBook/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn cargo_book() {
    let logs = MDBook::options()
        .max_log_level(tracing::Level::DEBUG)
        .load(BOOKS.join("cargo/src/doc"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/cargo/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_book() {
    let logs = MDBook::load(BOOKS.join("rust-book"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/book/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn nomicon() {
    let logs = MDBook::load(BOOKS.join("nomicon"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/nomicon/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_by_example() {
    let logs = MDBook::load(BOOKS.join("rust-by-example"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/rust-by-example/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_edition_guide() {
    let logs = MDBook::load(BOOKS.join("rust-edition-guide"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/edition-guide/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_embedded() {
    let logs = MDBook::load(BOOKS.join("rust-embedded"))
        .config(Config {
            hosted_html: Some("https://docs.rust-embedded.org/book/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_reference() {
    let logs = MDBook::load(BOOKS.join("rust-reference"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/reference/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rustc_dev_guide() {
    let logs = MDBook::load(BOOKS.join("rustc-dev-guide"))
        .config(Config {
            hosted_html: Some("https://rustc-dev-guide.rust-lang.org/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}
