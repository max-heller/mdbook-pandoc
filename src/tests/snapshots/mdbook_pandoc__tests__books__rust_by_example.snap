---
source: src/tests/books.rs
expression: logs
snapshot_kind: text
---
<<<<<<< HEAD
 INFO mdbook::book: Running the pandoc backend
 INFO mdbook_pandoc::pandoc::renderer: Running pandoc
||||||| parent of 8c10a4f (chore(deps)!: upgrade mdbook to v0.5.0)
 INFO mdbook::book: Running the pandoc backend    
 INFO mdbook_pandoc::pandoc::renderer: Running pandoc    
=======
 INFO mdbook_driver::mdbook: Running the pandoc backend    
 INFO mdbook_pandoc::pandoc::renderer: Running pandoc    
>>>>>>> 8c10a4f (chore(deps)!: upgrade mdbook to v0.5.0)
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pdf__src__std__result__question_mark.md' on $PAGE
  undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: There were undefined references.
[WARNING] Missing character: There is no ðŸ›ˆ (U+1F6C8) (U+1F6C8) in font NotoSerif:mode=node;script
[WARNING] Missing character: There is no ðŸ›ˆ (U+1F6C8) (U+1F6C8) in font NotoSerif:mode=node;script
 INFO mdbook_pandoc::pandoc::renderer: Wrote output to book/pdf/book.pdf
